Library/Homebrew/extend/os/mac/formula_cellar_checks.rb

```diff
@@ -88,16 +88,61 @@ def check_linkage
     end
   end
 
+  def check_cpuid_instruction
+    # TODO: add methods to `utils/ast` to allow checking for method use
+    return unless formula.path.read.include? "ENV.runtime_cpu_detection"
+    # Checking for `cpuid` only makes sense on Intel:
+    # https://en.wikipedia.org/wiki/CPUID
+    return unless Hardware::CPU.intel?
+
+    # macOS `objdump` is a bit slow, so we prioritise llvm's `llvm-objdump` (~5.7x faster)
+    # or binutils' `objdump` (~1.8x faster) if they are installed.
+    objdump   = Formula["llvm"].opt_bin/"llvm-objdump" if Formula["llvm"].any_version_installed?
+    objdump ||= Formula["binutils"].opt_bin/"objdump" if Formula["binutils"].any_version_installed?
+    objdump ||= which("objdump")
+    objdump ||= which("objdump", ENV["HOMEBREW_PATH"])
+    objdump ||= begin
+      # If the system provides no `objdump`, install binutils instead of llvm since
+      # binutils is smaller and has fewer dependencies.
+      ohai "Installing `binutils` for `cpuid` instruction check..."
+      safe_system HOMEBREW_BREW_FILE, "install", "binutils"
+      Formula["binutils"].opt_bin/"objdump"
+    end
+
+    keg = Keg.new(formula.prefix)
+    has_cpuid_instruction = false
+    keg.mach_o_files.each do |file|
+      has_cpuid_instruction = cpuid_instruction?(file, objdump)
+      break if has_cpuid_instruction
+    end
+    return if has_cpuid_instruction
+
+    "No `cpuid` instruction detected. #{formula} should not use `ENV.runtime_cpu_detection`."
+  end
+
   def audit_installed
     generic_audit_installed
     problem_if_output(check_shadowed_headers)
     problem_if_output(check_openssl_links)
     problem_if_output(check_python_framework_links(formula.lib))
     check_linkage
+    problem_if_output(check_cpuid_instruction)
   end
 
   def valid_library_extension?(filename)
     macos_lib_extensions = %w[.dylib .framework]
     generic_valid_library_extension?(filename) || macos_lib_extensions.include?(filename.extname)
   end
+
+  def cpuid_instruction?(file, objdump = "objdump")
```

Comment:
Not sure if this is the best way to do this, but the idea here is that I don't want to wait for `objdump` to finish dumping its output (which could take a while) when it has already produced a line containing a `cpuid` instruction.